Created sql script for creating the Race Table

anishprasad01 opened the pull request:
Added SQL Script for creating the Race Table. Also renamed LeaderboardScore to Race. Renamed ScoreId to RaceId.

anishprasad01 committed:
Created sql script for creating the Race Table

anishprasad01 committed:
Added IDENTITY(1,1) and UNIQUE constraints to RaceId in Create_Race_Table.sql

anishprasad01 committed:
Added constraint to RaceId to specify that it is the PRIMARY KEY. Removed UNIQUE constraint. Added comments to entire file

Chris-Johnston commented on PetGame/scripts/login.js:
```diff
@@ -0,0 +1,25 @@
+function tryLogin() {
```
Remove this file from the PR. It may have been copied over from the login flow PR by accident.

Chris-Johnston commented on PetGame/scripts/login.js.map:
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"login.js","sourceRoot":"","sources":["login.ts"],"names":[],"mappings":"AACA,SAAS,QAAQ;IAEb,IAAM,aAAa,GAAI,QAAQ,CAAC,cAAc,CAAC,UAAU,CAAsB,CAAC,KAAK,CAAC;IACtF,IAAM,aAAa,GAAI,QAAQ,CAAC,cAAc,CAAC,UAAU,CAAsB,CAAC,KAAK,CAAC;IACtF,IAAI,YAAY,GAAG,QAAQ,CAAC,cAAc,CAAC,SAAS,CAAyB,CAAC;IAE9E,IAAI,OAAO,GAAG,IAAI,cAAc,EAAE,CAAC;IACnC,OAAO,CAAC,IAAI,CAAC,MAAM,EAAE,YAAY,EAAE,IAAI,CAAC,CAAC ...
```
Remove this file.

Chris-Johnston commented on SQL/Create_Race_Table.sql:
```diff
@@ -0,0 +1,23 @@
+	--The pet's score in this race
+	[Score] INT NOT NULL,
+	--The timestamp is when the score is inserted into the table, after the race ends
+	[Timestamp] DATETIME NOT NULL,
```
Should default to `GETDATE()`

Chris-Johnston commented on SQL/Create_Race_Table.sql:
```diff
@@ -0,0 +1,23 @@
+
+SET QUOTED_IDENTIFIER ON
+GO
+
```
Please add a brief description of what this table is for.

Chris-Johnston commented on SQL/Create_Race_Table.sql:
```diff
@@ -0,0 +1,23 @@
+	--RaceId is the primary key, and for each Race, must be unique 
+	[RaceId] BIGINT IDENTITY(1,1) NOT NULL PRIMARY KEY,
+	--The pet's score in this race
+	[Score] INT NOT NULL,
```
Do we have constraints for the score yet? We should at least check that this is > 0.

Chris-Johnston commented on SQL/Create_Race_DummyData.sql:
```diff
@@ -0,0 +1,4 @@
+﻿USE [PetGame]
+GO
+
+INSERT INTO Race (RaceId, Score, Timestamp, PetId) VALUES (1234, 12, '12/12/2018 10:00:00 AM', 4657382);
```
Don't insert into the `RaceId` column. This should be done automatically.
Also, we should only insert for valid `PetId`.

anishprasad01 committed:
Made changes based on pull request comments:

Removed login.js and login.js.map
Added GETDATE() Default to Timestamp
Added Check that Score > 0
Modified insert statement in DummyData
Added table description comment

Chris-Johnston committed:
Merge branch 'add-sql-table-create-scripts' into create-race-table
